### PR TITLE
Add JOSM link

### PR DESCRIPTION
**Pull request by ericsoco**
/cc @mojodna

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -32,4 +32,24 @@ def potlatch_link(zoom, lon, lat, slug = nil)
...
+    dataRequest = "#{protocol}://#{domain}:#{port}/load_and_zoom?left=#{west}&right=#{east}&top=#{north}5&bottom=#{south}"
+
+    # load OSM tiles
+    if slug
+      tileRequest = "#{protocol}://#{domain}:#{port}/imagery?title=osm&type=tms&url=#{FieldPapers::TILE_BASE_URL}/snapshots/#{slug}/#{zoom}/#{lon}/#{lat}.png"
+    else
+      tileRequest = "#{protocol}://#{domain}:#{port}/imagery?title=osm&type=tms&url=https://a.tile.openstreetmap.org/#{zoom}/#{lon}/#{lat}.png"
+    end
+
+    return "data-data-request=#{dataRequest} data-tile-request=#{tileRequest}"
```
Comment (mojodna) [inline app/helpers/application_helper.rb]: Should the `data-*` values be quoted?

File: app/views/atlases/show.html.erb [hunk 2]
```diff
@@ -90,6 +90,8 @@
         provider: atlasProvider,
         pages: pages
       });
+
+      FP.setUpJOSMClickHandler("<%=escape_javascript _('Could not connect to JOSM. Make sure JOSM is running and that the Remote Control functionality is enabled in Preferences.') %>");
```
Comment (mojodna) [inline app/views/atlases/show.html.erb]: Minor style quibbles (spacing + consistent use of parens):

```
<%= escape_javascript(_('...')) %>
```

File: app/views/snapshots/show.html.erb [hunk 3]
```diff
@@ -54,6 +55,7 @@
     <% end %>
 
       FP.map.snapshot("map", "<%= tilejson_url(@snapshot) %>");
+      FP.setUpJOSMClickHandler("<%=escape_javascript _('Could not connect to JOSM. Make sure JOSM is running and that the Remote Control functionality is enabled in Preferences.') %>");
```
Comment (mojodna) [inline app/views/snapshots/show.html.erb]: Same style quibbles as above.

File: app/assets/javascripts/atlases.js [hunk 4]
```diff
@@ -1,2 +1,41 @@
 // Place all the behaviors and hooks related to the matching controller here.
```
Comment (mojodna) [inline app/assets/javascripts/atlases.js]: This should be in `app/assets/javascripts/application.js`, as it's needed by both the atlas and snapshot views (right now the snapshot view won't work).

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (ericsoco) [inline app/helpers/application_helper.rb]: Nope -- returning strings to be written directly into the markup.

Comment (mojodna) [inline app/helpers/application_helper.rb]: Doesn't the resulting markup look like this?

```html
<a id='josm_link' data-data-request=http://127.0.0.1:8111/load_and_zoom?... data-tile-request=http://127.0.0.1:8111/imagery>...</a>
```

(Where the URL attributes aren't quoted.)

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -32,4 +32,24 @@ def potlatch_link(zoom, lon, lat, slug = nil)
...
+    domain = "127.0.0.1"
+
+    # load OSM data and zoom to extents
+    dataRequest = "#{protocol}://#{domain}:#{port}/load_and_zoom?left=#{west}&right=#{east}&top=#{north}5&bottom=#{south}"
+
+    # load OSM tiles
+    if slug
+      tileRequest = "#{protocol}://#{domain}:#{port}/imagery?title=osm&type=tms&url=#{FieldPapers::TILE_BASE_URL}/snapshots/#{slug}/#{zoom}/#{lon}/#{lat}.png"
+    else
+      tileRequest = "#{protocol}://#{domain}:#{port}/imagery?title=osm&type=tms&url=https://a.tile.openstreetmap.org/#{zoom}/#{lon}/#{lat}.png"
```
Comment (mojodna) [inline app/helpers/application_helper.rb]: If no slug, use the provider associated with the atlas/snapshot (`atlas.get_provider_without_overlay`, `snapshot.provider`) with the placeholders changed to match JOSM's expectations (vs. defaulting to OSM).

**Pull request merged**